src/main/java/io/stargate/sgv2/jsonapi/service/bridge/executor/CollectionSettings.java

```diff
@@ -53,6 +59,52 @@ public static SimilarityFunction fromString(String similarityFunction) {
     }
   }
 
+  public static CollectionSettings getCollectionSettings(
+      TableMetadata table, ObjectMapper objectMapper) {
+    String collectionName = table.getName().asCql(true);
+    // get vector column
+    final Optional<ColumnMetadata> vectorColumn =
+        table.getColumn(DocumentConstants.Fields.VECTOR_SEARCH_INDEX_COLUMN_NAME);
+    boolean vectorEnabled = vectorColumn.isPresent();
+    // if vector column exist
+    if (vectorEnabled) {
+      final int vectorSize = ((VectorType) vectorColumn.get().getType()).getDimensions();
+      // get vector index
+      IndexMetadata vectorIndex = null;
+      Map<CqlIdentifier, IndexMetadata> indexMap = table.getIndexes();
+      for (CqlIdentifier key : indexMap.keySet()) {
+        if (key.asInternal().contains(DocumentConstants.Fields.VECTOR_SEARCH_INDEX_COLUMN_NAME)) {
```

Comment:
Should this use `equals()` instead of `contains()`?

Reply:
The string is like "collectionname_indexname". If we want to use `equal`, we have to concatenate the collection name and index name. So I think `contains` could be better?

Reply:
Hmmh. But it is comparing to index Column name?

But I agree, if this is true, then using `.startsWith()` or `.endsWith()` (if column name is always first thing, or last thing) would be better.